### PR TITLE
feat: require tenant-id header in management API routes

### PR DESCRIPTION
**Pull request by danielgent**
As discussed, this seems like the best way to support having multiple tenants under one Auth domain

### CODE REVIEW

File: src/routes/tsoa/users-mgmt.ts [hunk 1]
```diff
@@ -19,9 +19,14 @@ export class UsersMgmtController extends Controller {
     const { ctx } = request;
     const { env } = ctx;
 
+    const tenantId = request.headers["tenant-id"];
+
+    if (!tenantId) throw new Error("tenant-id header is required");
```
Comment (markusahlstrand) [inline src/routes/tsoa/users-mgmt.ts:24]: ohh.. a one liner :) I like to have the curlies and a new line to make it clear it's a conditional, but it's a personal preference

File: src/routes/tsoa/users-mgmt.ts [hunk 2]
```diff
@@ -19,9 +19,14 @@ export class UsersMgmtController extends Controller {
     const { ctx } = request;
     const { env } = ctx;
 
+    const tenantId = request.headers["tenant-id"];
```
Comment (markusahlstrand) [inline src/routes/tsoa/users-mgmt.ts:22]: We have a headers object in the constants (I think...) to avoid typos of headers.

Comment (markusahlstrand) [reply src/routes/tsoa/users-mgmt.ts:22]: TSOA handles headers for you which is pretty slick. Then it gets added to the docs correctly and is validated as well. Just add this above where the path is defined:
```
@Header("tenant-id) tenantId: string 
```

**Review by markusahlstrand — APPROVED** [review]
Looks good. Would be great to use the TSOA header so it's in the docs and validation.

File: src/routes/tsoa/users-mgmt.ts (hunk 2 above)
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts]: `We have a headers object in the constants` I'll make this change *but* everything builds without this (see previous commits)

File: src/routes/tsoa/users-mgmt.ts (hunk 1 above)
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts]: This line has gone anyway :sunglasses:

**Pull request merged**